src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +258,69 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
+        """
+        Switches to a different warehouse for a while, then switches back.
+        This is a no-op if the requested warehouse is already active.
+        If there is no default warehouse in the account, it will throw an error.
+        """
+        wh_result = self._execute_query(
+            f"select current_warehouse()", cursor_class=DictCursor
+        ).fetchone()
+        # If user has an assigned default warehouse, prev_wh will contain a value even if the warehouse is suspended.
+        prev_wh = wh_result["CURRENT_WAREHOUSE()"]
+
+        if new_wh is not None:
+            new_wh_id = to_identifier(new_wh)
+
+        if prev_wh is None and new_wh is None:
```

Comment:
this logic is a bit too complicated.
1) Why do we need to stay with prev_wh if new_wh is not defined? is it a valid use case that new_wh is None (especially when dealing with snow app run)? Shouldn't we just error?

2) too many combinations. I suggest to simplify as follows if possible:

a) if new_wh is None, error out.
b) proceed to switch to new_wh if different from prev_wh.
c) yield
d) if prev_wh was None, do nothing, otherwise, switch back to it.

Reply:
Some concerns raised in the previous comments:
1. It is possible that prev_wh is None, especially in the case when no default warehouse is assigned to a user, and hence `current_warehouse()` may return None. I do not have a way of verifying this as I do not have permissions to set up a new user in an account, but I do believe this is a case we need to account for since the bug ticket was created for a scenario like this when there is no fallback.
2. It is possible that new_wh is None. In our case, `package_warehouse` or `application_warehouse` could be None if user has not provided that warehouse in their snowflake.yml, config.toml and there is no default warehouse. In that case, the connection context would return a None for this property.
3. Given the two above, we do need to tackle their nullity accordingly.
But I am open to hearing if the assumptions in 1 and/or 2 are incorrect and therefore do not need to be handled.

Reply:
I'm with Michel, I think we can simplify it as suggested. Sent you a way to determine if warehouse is not set.